core/benches/bls_sigverify.rs

```diff
@@ -20,18 +20,54 @@ use {
         consensus_message::{Certificate, ConsensusMessage, VoteMessage},
         vote::Vote,
     },
-    std::{cell::RefCell, sync::Arc},
+    std::{
+        cell::RefCell,
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc,
+        },
+    },
 };
 
 const BENCH_SLOT: u64 = 70;
 // TODO(sam): use a small number for now to emulate the current test cluster
 const NUM_VALIDATORS: usize = 50;
 
+const NUM_PREGENERATED_BATCHES: usize = 500;
+
 struct BenchEnvironment {
     verifier: RefCell<BLSSigVerifier>,
     validator_keypairs: Arc<Vec<ValidatorVoteKeypairs>>,
 }
 
+struct PregeneratedBatches {
+    batches: Vec<Vec<PacketBatch>>,
+    counter: AtomicUsize,
+}
+
+impl PregeneratedBatches {
+    fn new<F>(env: &BenchEnvironment, generator: F) -> Self
+    where
+        F: Fn(&BenchEnvironment) -> Vec<PacketBatch>,
+    {
+        // Generate the batches upfront, moving expensive crypto ops out of the benchmark loop
+        let batches = (0..NUM_PREGENERATED_BATCHES)
+            .map(|_| generator(env))
+            .collect();
+        Self {
+            batches,
+            counter: AtomicUsize::new(0),
+        }
+    }
+
+    // Get the next batch in a round-robin fashion.
+    // We must clone because the verifier modifies the packets in place (e.g., setting discard flags).
+    fn next(&self) -> Vec<PacketBatch> {
+        let index = self.counter.fetch_add(1, Ordering::Relaxed) % self.batches.len();
+        self.batches[index].clone()
```

Comment:
- are we worried about a potentially expensive clone?  
- also just to make sure, the previous batches are getting dropped by the verifier or could this result in steadily increasing memory growth?